pytorch_lightning/utilities/types.py

```diff
@@ -49,8 +49,8 @@
 
 
 @runtime_checkable
-class _SupportsStateDict(Protocol):
-    """This class is used to detect if an object is stateful using `isinstance(obj, _SupportsStateDict)`."""
+class Stateful(Protocol):
```

Comment:
any reason why is it not protected anymore?

Reply:
I was thinking `Stateful` protocol should be considered "public" API, same level as `CheckpointHooks`

Reply:
can't say about `CheckpointHooks` and other hook classes now, possibly they will be made protected too since they are not used externally and are used only for the inheritance for public components like LM or LDM. But if they are targeted to be made protected too, we should make this protected now. Just to avoid 1 more possible deprecation.

cc @daniellepintz

Reply:
ah agree it's easier to elevate than deprecate so let's keep it protected for now

> else I think it will start appearing in API references in docs.

would we not want this?